### ISSUE TITLE
Find and search cannot be toggled off

### ISSUE THREAD

**Issue by sbauer322** [LightTable/LightTable#2285]
Find (Ctrl-F) and search (Ctrl-Shift-F) cannot be toggled off by the same command that opened it. This is inconsistent from other temporary widgets such as language docs (Ctrl-Shift-D), command pane (Ctrl-Space), and the console. 

Find can only be hidden via the command pane. Search can be closed by closing its tab. For consistency,  these should be toggle-able.

**Comment by Mouvedia** [comment]
I dunno about the desired behaviour from the user point of view in that scenario:
* search something which exists in the editor
* focus search field
* esc
* pmeta-f (`:find.show`)

I was surprised that the queried keyword wasn't automatically highlighted in the editor. Is this expected?

**Comment by ahoffer** [comment]
@sbauer322 
I am not sure what behavior ctrl+shift+f should exhibit. Ctrl_shift+f open the Searcher, which is a new tab. The new tab is made active. 

Should ctrl+shift+f close the Searcher tab and set focus to the tab on the left? This is what happens when you close the Searcher tab by clicking on the tab's "X" button.

Personally, I expected subsequent ctrl+shift+f invocations to open more Searchers tabs. Why limit users to one? I might want a Search tab for all references to "object/merge" strings and another Searcher tab for "<div>" strings.

What behavior satisfies the "principle of least surprise" for multiple ctrl+shift+f ?

**Comment by sbauer322** [comment]
Hmm, those are some good questions @ahoffer! 

There are a couple different paths to pursue, but in hindsight, this issue should not have included the `Ctrl-Shift-F` Search. As you rightly pointed out, since the tab has an "X" button, it is not necessary to be able to close the Search tab via `Ctrl-Shift-F`... additionally, whatever keybinding already exists to close a tab should be able to handle closing the Search tab already. This combination sufficiently handles closing the Search.

I am leaning towards narrowing the scope of this issue to just the `Ctrl-F` find bar and merge the related PR, if you don't mind. Another issue could then be opened for the Search to handle multiple Search independent results. Let me know how you want to proceed, @ahoffer.

---------------

(Just tossing this here since we are on the topic of Search and it can be used as the basis of the new Search issue mentioned above)

Regarding what the behavior should be for the Search, I can't say I have the right answer, but let's pick something and give it a try. Worst case we just change/revert the behavior. That said, I do not know offhand how difficult it would be to implement some of those features (especially additional search tabs).

The following are a few possible paths, but I am open to alternatives.

## Leave as is

For me, the least surprising (and simplest) things would be the current behavior:

- When not focused on the search tab, focus on the Search tab
- When focused on the search tab, reset the cursor to the 'Search' field
- The `Tab` button cycles through fields within the Search tab

This is the least surprising behavior for me due to other editors already supporting this exact behavior, most notably Atom.

## Multiple search tabs

If we wanted to support additional search tabs then perhaps the behavior should be:

- When not focused on a search tab, focus on the last active Search tab (very similar to current behavior)
- When focused on a search tab, open a new Search tab
- When focused on a search tab, the `Tab` button cycles through fields within the active Search tab
- Add a command and keybinding to allow for cycling through the Search tabs rather than opening a another

## Multiple collapsible results in one search tab (my personal favorite)

This would dovetail nicely with issue #2240, which asks for collapsible individual file results. Basically, we make each entire search collapsible (and closable). Notepad++ already does something similar. This easily gives access to previous results while minimizing the number of tabs opened.

I feel this option would be the most beneficial while still remaining familiar to current users.

**Comment by ahoffer** [comment]
Agreed. Merge [PR 2330](https://github.com/LightTable/LightTable/pull/2330). When issue #2240 is resolved, do the same thing for search. I like Notepad++ collapsable/closable search results as well.
